hal/hal_uv.c

```diff
@@ -23,25 +23,35 @@
 #include "hal_uv_priv.h"
 #include <uv.h>
 
-static adb_context_uv_t g_adbd_context;
-
 /****************************************************************************
  * HAL Public Functions
  ****************************************************************************/
 
 adb_context_t* adb_hal_create_context(void) {
-    adb_context_uv_t *adbd = &g_adbd_context;
+    adb_context_uv_t *adbd = malloc(sizeof(adb_context_uv_t));
```

Comment:
what is the usecase of having multiple contexts?
can't we fit everything into the same event loop?

Reply:
no, just make the code simple and avoid consuming memory before launch adbd.